src/server/frontend/wayland/wl_mir_window.cpp

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright © 2015-2018 Canonical Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Authored by: Christopher James Halse Rogers <christopher.halse.rogers@canonical.com>
+ */
+
+#include "wl_mir_window.h"
+
+#include "wayland_connector.h"
```

Comment:
The only thing used from wayland_connector.h is session_for_client(), surely there's a better place to put that so we can lose this dependency?

Reply:
As far as I can tell, there currently is not (at least not any place that makes logical sense). Perhaps we should have wayland_utils.h or something along those lines? run_unless could also go there (although in that case, I have ideas about it being moved to the auto generated code down the road).